src/createAuthScheme.js

```diff
@@ -65,12 +65,66 @@ module.exports = function createAuthScheme(authFun, authorizerOptions, funName,
           return onError(err);
         }
 
-        const onSuccess = policy => {
-        // Validate that the policy document has the principalId set
+        const onInvalidPolicyFormat = invalid => {
+          serverlessLog(`Authorization failed due to an invalid policy: (λ: ${authFunName})`, `Invalid ${invalid} entry`);
+
+          return reply(Boom.unauthorized('Unauthorized'));
+        };
+
+        // Make sure that the policy we receive is actually valid
+        const validatePolicyFormat = policy => {
+          let valid = true;
+
           if (!policy.principalId) {
-            serverlessLog(`Authorization response did not include a principalId: (λ: ${authFunName})`, err);
+            onInvalidPolicyFormat('principalId');
+            valid = false;
+          }
+
+          const policyDocument = policy.policyDocument;
+
+          // If they supply a policy document check its validitiy
+          // These are the absolute basics needed for a policy document
+          // outlined here: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html
+          if (policyDocument) {
+            // Only 2 policy documen versions exist 
+            if (!policyDocument.Version || (policyDocument.Version !== '2012-10-17' && policyDocument.Version !== '2008-10-17')) {
+              onInvalidPolicyFormat('version');
+              valid = false;
+            }
+
+            if (!policyDocument.Statement || !(policyDocument.Statement instanceof Array)) {
+              onInvalidPolicyFormat('statement');
+              valid = false;
+            }
+
+            if (!(policyDocument[0] instanceof Object)) {
```

Comment:
@PorterK It seems that you forgot to use the Statement field on all your if statements. As a result, the current version (3.24.0) doesn't work. @dherault